lib/spack/spack/cmd/create.py

```diff
@@ -125,11 +129,17 @@ def install(self, spec, prefix):
         make('install')"""
 
     url_line = '    url      = "{url}"'
+    git_line = '    git      = "{git_url}"'
 
-    def __init__(self, name, url, versions):
+    def __init__(self, name, url, versions, git_url=None):
         super(PackageTemplate, self).__init__(name, versions)
 
         self.url_def = self.url_line.format(url=url)
+        if git_url is not None:
```

Comment:
Perhaps better to also not allow a blank string:

```suggestion
        if git_url:
```